library/proc_macro/src/bridge/mod.rs

```diff
@@ -175,6 +175,30 @@ macro_rules! with_api {
     };
 }
 
+// Similar to `with_api`, but only lists the types, and they are divided into
+// the two storage categories.
+macro_rules! with_api_types {
+    ($m:ident) => {
+        $m! {
+            'owned:
+            FreeFunctions,
+            TokenStream,
+            TokenStreamBuilder,
+            TokenStreamIter,
+            Group,
+            Literal,
+            SourceFile,
+            MultiSpan,
+            Diagnostic,
+
+            'interned:
+            Punct,
+            Ident,
+            Span,
+        }
+    };
+}
```

Comment:
This isn't great and it would be useful to have this information either in `with_api!` entirely or not at all there.

Reply:
I think it's a clear improvement over the existing code. E.g. this macro is right next to `with_api!`, previously this information was in the `define_handles!` macro in a different file.